### ISSUE TITLE
Add pagination support

### ISSUE THREAD

**Issue by Barbapapazes** [nuxt/content#1918]
l'm currently building a large application using Content. In order to have a better UX, I'm looking for a way to paginate data.

Like `find` or `findOne` we could have `paginate`. (I can simulare this using skip and limit but I don't have a way to count all).

```ts
const data = queryContent('/categories').paginate(<page:number>, <limit:number>)
```

Behind the scene, Content fetch data, count and create pagination to return 

```ts
{
  data: <data: any[]>
  meta: {
    total: <number>,
    perPage: <number>,
    currentPage: <number>,
    lastPage: <number>,
    firstPage: <number>
  }
}
```

Reference:
  - https://docs.adonisjs.com/reference/database/query-builder#pagination
  - https://docs.adonisjs.com/guides/database/pagination

**Comment by Barbapapazes** [comment]
In the same time, a count method could be awesome.

**Comment by bordermedia** [comment]
+1 for adding a count method.  Currently have to run a `queryContent` for all content to get total, then perform a pagination calc, then re-run `queryContent` with skip and limit.

**Comment by wokalek** [comment]
```js
// fetch the next 5 articles
const articles = await queryContent('articles')
    .skip(5)
    .limit(5)
    .find()
```

**Comment by Barbapapazes** [comment]
> ```js
> // fetch the next 5 articles
> const articles = await queryContent('articles')
>     .skip(5)
>     .limit(5)
>     .find()
> ```

Yes but No because with this approach, you won't be able to know when you reach the last page and you will have to have another request to have the number of articles. The request to get the number of articles, you will transfert all content, not just a simple number.

**Comment by wokalek** [comment]
> > ```js
> > // fetch the next 5 articles
> > const articles = await queryContent('articles')
> >     .skip(5)
> >     .limit(5)
> >     .find()
> > ```
> 
> Yes but No because with this approach, you won't be able to know when you reach the last page and you will have to have another request to have the number of articles. The request to get the number of articles, you will transfert all content, not just a simple number.

Yeah, I do not argue, it is necessary in the nuxt/content

**Comment by dixhuit** [comment]
Related:

- https://github.com/nuxt/content/issues/1389
- https://github.com/nuxt/content/issues/378
- https://github.com/nuxt/content/issues/293
- https://github.com/nuxt/content/issues/384
